core/zip/inc/Compression.h

```diff
@@ -39,56 +39,57 @@ namespace ROOT {
 ///   [207 - 208]
 ///  - LZ4 is recommended to be used with compression level 4 [404]
 
+class RCompressionSetting {
```

Comment:
For now I would keep it as a namespace (as long as we don't need to forward declare it anywhere)

Reply:
I thought we wanted to keep it as a class in order to flesh out the implementation in 6.18?

Reply:
Either way.   I was thinking (assuming we don't need any forward declaration) that we could have it as a namespace now, as it is not usable as struct yet, and change it to a struct when we are implementing the object.  This change is just a recommendation (and is not necessary/up for discussion).

Reply:
Hm - that would keep source code compatibility, right?  It would cause ABI compatibility (any concerns with API?), but is it correct we don't care between 6.16 and 6.18?

Reply:
> that would keep source code compatibility, right?

Yes, unless somebody feel the need to forward declare it ... 

> It would cause ABI compatibility (any concerns with API?), but is it correct we don't care between 6.16 and 6.18?

Yes we make no guarantee on ABI compatibility (actually au contraire) between major versions (and even for patch release we are often not doing it either).